packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -30,6 +30,10 @@ class AnalyzeCommand extends FlutterCommand {
         valueHelp: 'file',
         help: 'Also output the results to a file. This is useful with --watch '
               'if you want a file to always contain the latest results.');
+    argParser.addFlag('successful-if-correct',
```

Comment:
I'd prefer to avoid adding additional flags to `flutter analyze` - we already have a large UI surface area here, as well as for the other `flutter` tools commands.

If you do want to differentiate between successful runs that found analysis errors, and tool crashes, I would look at the `runner.dart` file - we should report crashes as a exit code of 64, but it looks like we're not super consistent in that file.